### PR TITLE
Fairness methods

### PR DESCRIPTION
**Pull request by MargaretBassney**

### CODE REVIEW

**Comment by MargaretBassney** [comment]
creating pull request for reviewing, I will just copy and paste it in main when its ready

File: term_paper/term_paper.Rmd [hunk 1]
```diff
@@ -447,23 +447,133 @@ educ_perc = (df2.groupby(['Education'])['has_access']
...
+
+Individual fairness metrics measure how similarly the model predicts for similar observations. Will two very similar people receive the same classification? Individual fairness metrics contradict group fairness metrics. When accounting for imbalanced predictions between groups, the within group fairness can suffer[@kypraiou_what_2021]. In the process of satisfying group metrics, two similar subjects only differing by sex, may be classified differently [@binns2020apparent; @mehrabi2021survey; @caton2020fairness; @zhou2022bias]. The individual metrics we explore are general entropy error and consistency score. We can measure both individual and group fairness with the between group general entropy error metric. 
+
+What happens when we remove the sensitive attributes in our data and then run the model? Will the model still be biased against protected groups? In most cases, no. There are often variables that remain in the data that act as pseudo substitutes for the protected attribute [@zhou2022bias]. For example, if race was excluded from the model but the variable zip code remained. Zip code can act as a stand in for race in regions where people are segregated by race.
```
Comment (rporta23) [inline term_paper/term_paper.Rmd:475]: I think you meant "In most cases, yes" on line 475 instead of no. You explain afterward how it will still be biased even if the protected attribute is not directly included.

File: term_paper/term_paper.Rmd [hunk 2]
```diff
@@ -447,23 +447,133 @@ educ_perc = (df2.groupby(['Education'])['has_access']
...
+
+What happens when we remove the sensitive attributes in our data and then run the model? Will the model still be biased against protected groups? In most cases, no. There are often variables that remain in the data that act as pseudo substitutes for the protected attribute [@zhou2022bias]. For example, if race was excluded from the model but the variable zip code remained. Zip code can act as a stand in for race in regions where people are segregated by race. 
+
+In our case we are focusing on the protected attribute gender. The other variables in this data set are education, age, and all the financial variables. If any of these variables are segregated by gender, by inherit gender bias in society, then simply removing the gender variable wont solve anything.
```
Comment (rporta23) [inline term_paper/term_paper.Rmd:477]: you for got the apostrophe in "won't", but I would recommend changing that to "will not" anyway to keep formal academic tone.

File: term_paper/term_paper.Rmd [hunk 3]
```diff
@@ -447,23 +447,133 @@ educ_perc = (df2.groupby(['Education'])['has_access']
...
+
+
+**Accuracy**
+Accuracy is a measure of how many classifications our model predicts correctly compared to all the predictions. The ratio of correctly predicted classifications to all the predictions. Accuracy cannot tell us if the predictions are equally correct across positives and negatives [@juba2019precision; @gupta2021recall]. 55% of the people in our data set don't have access to emergency funds. As long as our model predicts negatives more than half the time, we can get a good accuracy. However, our model will lose the ability to accurately predict positives. It is important to consider accuracy along with precision and recall so we can more fully understand how our model is classifying people.
```
Comment (rporta23) [inline term_paper/term_paper.Rmd:481]: I think it may not be necessary to write "Accuracy" twice-- not sure if you were trying to make a heading or if you were just trying to bold, it, but double check the formatting on that. Same thing with precision and recall.

File: term_paper/term_paper.Rmd (hunk 1 above)
Comment (rporta23) [reply term_paper/term_paper.Rmd:475]: Also, it may be better to rephrase this as something like, "Although it may seem an intuitive approach to simply remove the protected attribute from the data in order to make the algorithm unbiased, this is often not an effective approach to reduce bias." Given that this is an academic paper, I think we want to try to directly state our points and don't need to have rhetorical questions to try to keep the reader engaged.

**Review by rporta23 — COMMENTED** [review]
Besides the small wording/formatting comments that I left, I think it looks great and effectively clarifies all of the metrics and techniques we are using. Feel free to merge it in after you make the small changes.
